src/Compilers/Core/Portable/SourceGeneration/Nodes/ISyntaxInputNode.cs

```diff
@@ -3,22 +3,26 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Collections.Generic;
 using System.Threading;
 using Microsoft.CodeAnalysis.Collections;
 
 namespace Microsoft.CodeAnalysis
 {
     internal interface ISyntaxInputNode
```

Comment:
>ISyntaxInputNode [](http://example.com/codeflow?start=23&length=16)

Is this interface still needed if there's just a single implementation?

Reply:
Annoyingly yes, because the nodes themselves are generic. We need a way to put them all in a list and ask for their builders when we don't know what their type is upfront.

Appreciate any suggestions to make this better though.

Reply:
Perhaps I should just make it base type instead of an interface?

Reply:
Ended up making it a base type.